internal/exec/terraform_generate_varfiles.go

```diff
@@ -0,0 +1,153 @@
+package exec
+
+import (
+	"fmt"
+	c "github.com/cloudposse/atmos/pkg/config"
+	s "github.com/cloudposse/atmos/pkg/stack"
+	u "github.com/cloudposse/atmos/pkg/utils"
+	"github.com/spf13/cobra"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// ExecuteTerraformGenerateVarfilesCmd executes `terraform generate varfiles` command
+func ExecuteTerraformGenerateVarfilesCmd(cmd *cobra.Command, args []string) error {
+	flags := cmd.Flags()
+
+	fileTemplate, err := flags.GetString("file-template")
+	if err != nil {
+		return err
+	}
+
+	stacksCsv, err := flags.GetString("stacks")
+	if err != nil {
+		return err
+	}
+	var stacks []string
+	if stacksCsv != "" {
+		stacks = strings.Split(stacksCsv, ",")
+	}
+
+	componentsCsv, err := flags.GetString("components")
+	if err != nil {
+		return err
+	}
+	var components []string
+	if componentsCsv != "" {
+		components = strings.Split(componentsCsv, ",")
+	}
+
+	format, err := flags.GetString("format")
+	if err != nil {
+		return err
+	}
+	if format != "" && format != "yaml" && format != "json" {
+		return fmt.Errorf("invalid '--format' flag '%s'. Valid values are 'json' (default) and 'yaml'", format)
+	}
+	if format == "" {
+		format = "json"
+	}
+
+	return ExecuteTerraformGenerateVarfiles(fileTemplate, format, stacks, components)
+}
+
+// ExecuteTerraformGenerateVarfiles generates varfiles for all terraform components in all stacks
+func ExecuteTerraformGenerateVarfiles(fileTemplate string, format string, stacks []string, components []string) error {
+	var configAndStacksInfo c.ConfigAndStacksInfo
+	stacksMap, err := FindStacksMap(configAndStacksInfo, false)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println()
+
+	for stackConfigFileName, stackSection := range stacksMap {
+		if componentsSection, ok := stackSection.(map[any]any)["components"].(map[string]any); ok {
+			if terraformSection, ok := componentsSection["terraform"].(map[string]any); ok {
+				for componentName, compSection := range terraformSection {
+					if componentSection, ok := compSection.(map[string]any); ok {
+						// Find all derived components of the provided components
+						derivedComponents, err := s.FindComponentsDerivedFromBaseComponents(stackConfigFileName, terraformSection, components)
+						if err != nil {
+							return err
+						}
+
+						if len(components) == 0 || u.SliceContainsString(components, componentName) || u.SliceContainsString(derivedComponents, componentName) {
```

Comment:
Some of the below lines seemed to be shared with `internal/exec/atlantis_generate_repo_config.go`. Can we create a function or a few functions to DRY it?